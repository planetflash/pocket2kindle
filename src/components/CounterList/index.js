import React from "react";

import Counter from "components/Counter";
import Button from "components/Button";

const createRange = num => Array.from(Array(num).keys());

class CounterList extends React.Component {
  state = {
    numCounters: 2
  };

  addCounter = () =>
    this.setState(prevState => ({
      numCounters: prevState.numCounters + 1
    }));

  makeIncrementer = amount => () =>
    this.setState(prevState => ({
      count: prevState.count + amount
    }));

  renderCounter = index => {
    return (
      <li key={index}>
        <Counter />
      </li>
    );
  };

  render() {
    const countersArray = createRange(this.state.numCounters);

    return (
      <div>
        <Button onClick={this.addCounter} text={"Add Counter"} />

        <ul>{countersArray.map(num => this.renderCounter(num))}</ul>
      </div>
    );
  }
}

export default CounterList;
