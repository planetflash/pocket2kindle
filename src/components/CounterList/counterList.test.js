import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import CounterList from "./";

describe("CounterList component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<CounterList />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render two counters bt default", () => {
    const wrapper = shallow(<CounterList />);
    const counters = wrapper.find("Counter");

    expect(counters.length).toEqual(2);
  });

  it("can add more counters when we click the button", () => {
    const wrapper = shallow(<CounterList />);
    const button = wrapper.find("Button");
    button.simulate("click");
    const counters = wrapper.find("Counter");

    expect(counters.length).toEqual(3);
  });
});
