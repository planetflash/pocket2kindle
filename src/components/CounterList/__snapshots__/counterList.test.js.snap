// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterList component matches the snapshot 1`] = `
<div>
  <div>
    <a
      className="src-components-Button-___style__btn___2qMxX"
      onClick={[Function]}
    >
      <span
        className="text"
      >
        Add Counter
      </span>
    </a>
  </div>
  <ul>
    <li>
      <div>
        <p>
          Current count: 
          0
        </p>
        <div>
          <a
            className="src-components-Button-___style__btn___2qMxX"
            onClick={[Function]}
          >
            <span
              className="text"
            >
              +++
            </span>
          </a>
        </div>
        <div>
          <a
            className="src-components-Button-___style__btn___2qMxX"
            onClick={[Function]}
          >
            <span
              className="text"
            >
              ---
            </span>
          </a>
        </div>
      </div>
    </li>
    <li>
      <div>
        <p>
          Current count: 
          0
        </p>
        <div>
          <a
            className="src-components-Button-___style__btn___2qMxX"
            onClick={[Function]}
          >
            <span
              className="text"
            >
              +++
            </span>
          </a>
        </div>
        <div>
          <a
            className="src-components-Button-___style__btn___2qMxX"
            onClick={[Function]}
          >
            <span
              className="text"
            >
              ---
            </span>
          </a>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
